File: src/ClientApp/src/app/models/Exercise.ts
import { MuscleGroup } from './MuscleGroup';

export interface IExercise {
    id: number;
    name: string;
    description: string;
    videoLink: string;
    muscleGroups: MuscleGroup[];

    isCompound(): boolean;
}

export class Exercise implements IExercise {
    constructor(id: number, name: string, description: string, videoLink: string, muscleGroups: MuscleGroup[]) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.videoLink = videoLink;
        this.muscleGroups = muscleGroups;
    }

    id: number;
    name: string;
    description: string;
    videoLink: string;
    muscleGroups: MuscleGroup[];

    isCompound() {
        if (this.muscleGroups && this.muscleGroups.length > 1) { return true; }
        return false;
    }
}

File: src/Models/Entities/Exercise.cs
using Newtonsoft.Json;
using System.Collections.Generic;
using System.Linq;

namespace WorkoutTracker.Models.Entities
{
    /// <summary>
    /// Exercise model.
    /// </summary>
    public class Exercise : BaseModel
    {
        /// <summary>
        /// The name of the exercise.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// A description of the exercise.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// A link to the exercise.
        /// </summary>
        public string VideoLink { get; set; }

        [JsonIgnore]
        public virtual ICollection<ExerciseMuscleGroupJoin> ExerciseMuscleGroups { get; set; } = 
            new List<ExerciseMuscleGroupJoin>();

        /// <summary>
        /// Join entities between exercrise and muscle groups.
        /// </summary>
        public ICollection<MuscleGroup> MuscleGroups => ExerciseMuscleGroups.Select(x => x.MuscleGroup).ToList();
    }
}
